components/script/dom/shadowroot.rs

```diff
@@ -326,9 +326,24 @@ impl VirtualMethods for ShadowRoot {
         let shadow_root = self.upcast::<Node>();
 
         shadow_root.set_flag(NodeFlags::IS_CONNECTED, context.tree_connected);
-        for node in shadow_root.children() {
+
+        for node in shadow_root.traverse_preorder(ShadowIncluding::No) {
```

Comment:
This should be `ShadowIncluding::Yes`, otherwise children inside two nested shadow roots won't be bound to the tree